src/Aquifer.API/Endpoints/Resources/Content/PullFromPublisherReview/Endpoint.cs

```diff
@@ -0,0 +1,79 @@
+using Aquifer.API.Common;
+using Aquifer.API.Services;
+using Aquifer.Data;
+using Aquifer.Data.Entities;
+using FastEndpoints;
+using Microsoft.EntityFrameworkCore;
+
+namespace Aquifer.API.Endpoints.Resources.Content.PullFromPublisherReview;
+
+public class Endpoint(AquiferDbContext dbContext, IUserService userService, IResourceHistoryService historyService) : Endpoint<Request>
+{
+    public override void Configure()
+    {
+        Post(
+            "/resources/content/pull-from-publisher-review",
+            "/resources/content/{ContentId}/pull-from-publisher-review");
+        Permissions(PermissionName.AssignContent, PermissionName.AssignOverride, PermissionName.AssignOutsideCompany);
+    }
+
+    public override async Task HandleAsync(Request request, CancellationToken ct)
+    {
+        var user = await userService.GetUserFromJwtAsync(ct);
+        var userToAssign = await dbContext.Users
+            .AsTracking()
+            .SingleOrDefaultAsync(u => u.Id == request.AssignedUserId && u.Enabled, ct);
+        var permissions = ResourceStatusHelpers.GetAssignmentPermissions(userService);
+
+        await ResourceStatusHelpers.ValidateReviewerAndAssignedUser<Request>(request.AssignedUserId, request.AssignedReviewerUserId,
+            dbContext, user, permissions, ct);
+
+        var contentIds = request.ContentId is not null ? [(int)request.ContentId] : request.ContentIds!;
+
+        var draftVersions = await ResourceStatusHelpers.GetDraftVersions<Request>(contentIds, dbContext, ct);
```

Comment:
Seems like we're making an assumption about the statuses that this is allowed to be. I think we should throw a validation error if any of the `draftVersions` aren't in the correct original status.